pramen-py/src/pramen_py/metastore/reader.py

```
@@ -119,50 +196,58 @@ def get_latest_available_date(
+        if not dates_list:
+            logger.error(
+                f"The directory does not contain partitions by "
+                f"'{metastore_table.info_date_settings.column}': {metastore_table.path}"
             )
+            raise ValueError("No partitions are available")
 
-        files = self.fs_utils.list_files(
-            table.path,
-            file_pattern=f"{table.info_date_settings.column}=*",
-        )
-        logger.debug(
-            f"The following files are in the {table.path}:\n"
-            f"{chr(10).join(files)}"
+        filtered_dates_list = self._filter_info_dates_by_until(
+            dates_list, until  # type: ignore
         )
 
-        try:
-            latest_date = max(
-                filter(
-                    before_until,
-                    map(
-                        extract_dates,
-                        files,
+        if not filtered_dates_list:
+            str_date_list = list(
+                map(
+                    lambda date: convert_date_to_str(
+                        date, metastore_table.info_date_settings.format
                     ),
+                    filtered_dates_list,
                 )
             )
-        except ValueError as err:
-            raise ValueError(
-                f"No partitions are available for the given {table_name}. "
-                f"The content of a table path is:\n"
-                f"{chr(10).join(files)}\n"
+            logger.error(
+                f"No partitions are available for the given '{metastore_table.name}'.\n"
+                f"The table is available for the following dates:\n"
+                f"{str_date_list}\n"
                 f"Only partitions earlier than {str(until)} might be included."
-            ) from err
-        else:
-            logger.info(f"Latest date for {table_name} is {latest_date}")
-            return latest_date
+            )
+            raise ValueError("No partitions are available")
```

Same here - you can raise an error only, but it can contain the more detailed message.